### PR TITLE
Undo and redo implementation.

### PR DESCRIPTION
**Pull request by dtjanaka**

### CODE REVIEW

File: src/main/ngapp/accio/src/app/editor/mask-controller.ts [hunk 1]
```diff
@@ -1,27 +1,70 @@
...
+  private pPresent: number = -1;
 
-  constructor() { }
+  constructor(private mask: Set<number> = new Set()) {}
 
-  // Cannot access mask directly, so that a mask can never be accidentally
-  // altered without also changing its history.
-  getMask(): Set<number> {
-    // Change this implementation if undo/redo design is changed
+  public getMask(): Set<number> {
```
Comment (charmychancey) [inline src/main/ngapp/accio/src/app/editor/mask-controller.ts:20]: Can you add a comment here to clarify that you're getting the "present" mask, or which version of the mask for this function.
Other than that, lgtm

File: src/main/ngapp/accio/src/app/editor/set-operator.spec.ts [hunk 2]
```diff
@@ -1,25 +1,29 @@
 import { SetOperator } from './set-operator';
 
 describe('SetOperator', () => {
- // TODO
- const setA = new Set([1, 2, 3, 4]);
- const setB = new Set([3, 4, 5, 6]);
+  // TODO
```
Comment (dtjanaka) [inline src/main/ngapp/accio/src/app/editor/set-operator.spec.ts:4]: @charmychancey what is this TODO for?

Comment (charmychancey) [reply src/main/ngapp/accio/src/app/editor/set-operator.spec.ts:4]: I forgot to remove it, you can go ahead and do so

**Comment by dtjanaka** [comment]
merge-branch contains the most up-to-date undo/redo code

**Pull request closed without merging**